### PR TITLE
Test cases for DeviceTypeManagementService.

### PR DESCRIPTION
**Pull request by menakaj**
## Automation tests
 - Unit tests 
   > Code Coverage 100%

### CODE REVIEW

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceTypeManagementServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+                .mock(DeviceManagementProviderServiceImpl.class, Mockito.RETURNS_MOCKS);
+        this.deviceTypeManagementService = new DeviceTypeManagementServiceImpl();
+    }
+
+    @Test(description = "Testing for existing device types.")
+    public void testExistingDeviceType() throws Exception {
+        PowerMockito.stub(PowerMockito.method(DeviceMgtAPIUtils.class, "getDeviceManagementService"))
+                .toReturn(this.deviceManagementProviderService);
+        Response response = this.deviceTypeManagementService.getDeviceTypes("");
+        Assert.assertNotNull(response);
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceTypeManagementServiceTest.java]: Shall we add a next parameter, to specify why the test case is failing? Can you update in all the test cases

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceTypeManagementServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        DeviceTypeMetaDefinition deviceTypeMetaDefinition = new DeviceTypeMetaDefinition();
+        deviceTypeMetaDefinition.setClaimable(true);
+        deviceTypeMetaDefinition.setDescription(DEVICE_TYPE_DESCRIPTION);
+
+        PushNotificationConfig pushNotificationConfig =
+                new PushNotificationConfig(TEST_DEVICE_TYPE, true, null);
+        deviceTypeMetaDefinition.setPushNotificationConfig(pushNotificationConfig);
+
+        deviceType.setDeviceTypeMetaDefinition(deviceTypeMetaDefinition);
+
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceTypeManagementServiceTest.java]: Remove the un-wanted extra lines

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/util/DeviceMgtAPITestUtils.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+package org.wso2.carbon.device.mgt.jaxrs.service.impl.util;
+
+import org.wso2.carbon.device.mgt.common.push.notification.PushNotificationConfig;
+import org.wso2.carbon.device.mgt.common.type.mgt.DeviceTypeMetaDefinition;
+import org.wso2.carbon.device.mgt.core.dto.DeviceType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DeviceMgtAPITestUtils {
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/util/DeviceMgtAPITestUtils.java]: Shall we add a class level comment

**Review by Megala21 — CHANGES_REQUESTED** [review]

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceTypeManagementServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+import static org.mockito.MockitoAnnotations.initMocks;
+
+/**
+ * This class holds the unit tests for the class {@link DeviceTypeManagementService}
+ */
+@PowerMockIgnore("javax.ws.rs.*")
+@SuppressStaticInitializationFor({"org.wso2.carbon.device.mgt.jaxrs.util.DeviceMgtAPIUtils",
+        "org.wso2.carbon.context.CarbonContext"})
+@PrepareForTest({DeviceMgtAPIUtils.class, MultitenantUtils.class, CarbonContext.class,
```
Comment (madawas) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceTypeManagementServiceTest.java]: Is it necessary to prepare MultitenantUtils.class and CarbonContext.class here?

**Review by Megala21 — APPROVED** [review]

**Pull request merged**